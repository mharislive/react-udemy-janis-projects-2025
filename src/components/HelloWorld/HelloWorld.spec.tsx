import { render, screen } from "@testing-library/react";
import HelloWorld from "./HelloWorld";

describe("HelloWorld", () => {
  it("renders heading with text", () => {
    render(<HelloWorld />);
    expect(screen.getByRole("heading", { level: 1 })).toBeInTheDocument();
  });

  it("should render correct text message of hello World", () => {
    render(<HelloWorld />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
      "Hello World"
    );
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<HelloWorld />);
    expect(asFragment()).toMatchSnapshot();
  });
});
