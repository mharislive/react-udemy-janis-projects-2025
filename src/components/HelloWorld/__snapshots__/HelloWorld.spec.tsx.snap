// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    Hello World
  </h1>
</DocumentFragment>
`;
